src/main/java/com/netgrif/application/engine/petrinet/domain/Component.java

```diff
@@ -40,4 +41,13 @@ public Component(String name, Map<String, String> properties, List<Icon> optionI
         this.properties = properties;
         this.optionIcons = optionIcons;
     }
+
+    @Override
+    public Component clone() {
+        Component clone = new Component();
+        clone.setName(this.name);
+        clone.setProperties(new HashMap<>(this.properties));
+        clone.setOptionIcons(this.optionIcons.stream().map(Icon::clone).collect(Collectors.toList()));
```

Comment:
Can optionIcons be null? Would be better to check for nullpointer

Reply:
Corrected